pkg/model/api/v1/dashboard.go

```diff
@@ -212,24 +214,21 @@ func (d *Dashboard) verifyAndSetJSONReferences() error {
 	return nil
 }
 
-func (d *Dashboard) checkAndSetRef(ref *common.JSONRef) error {
-	// ref.Path should like that [ "spec", "panels", <name> ].
-	// So if the array is not equal to three then the reference is wrong.
-	if len(ref.Path) != 3 {
+func checkAndSetRef(ref *common.JSONRef, panels map[string]*Panel) error {
+	// ref.Path should be like [ "spec", "panels", <name> ]
+	if len(ref.Path) != len(panelsRefPath)+1 {
```

Comment:
I bugged 2 sec on "what the hell is that ``panelsRefPath`` variable". Maybe it's not a big performance issue if you redeclare the list inside the function? WDYT?